src/time-picker/time-picker.jsx

```diff
@@ -108,6 +108,17 @@ const TimePicker = React.createClass({
       muiTheme: this.context.muiTheme || getMuiTheme(),
     };
   },
+  
+  componentWillReceiveProps(nextProps, nextContext) {
+    const newState = this.state;
+    if (nextContext.muiTheme) {
+      newState.muiTheme = nextContext.muiTheme; 
+    }
+    if (!Date.isEqualTime(this.state.time, nextProps.defaultTime)) {
+      newState.time = nextProps.defaultTime;
```

Comment:
I think that the `defaultTime` should only be used during the first render. I would add a new `value` property for a controlled use case.